server/controllers/userController.js

```diff
@@ -0,0 +1,163 @@
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @returns {object} Object containing token to the user
+   * @memberof UsersController
+   */
+  static async signUp(req, res, next) {
+    try {
+      const data = req.body;
+      const { password } = data;
+      data.password = await encryptPassword(password);
+      const user = await userModel.create(data);
+      const token = await generateToken(user);
+
+      return res.status(201).json({
+        status: 'success',
+        data: { token }
+      });
+    } catch (err) {
+      return next(err);
+    }
+  }
+
+  /**
+   *
+   * Login Middleware - Logs user into the application
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @returns {object} Object containing token to the user
+   * @memberof UsersController
+   */
+  static async login(req, res, next) {
+    try {
+      const data = req.body;
+      const { userLogin, password } = data;
+      // Login with username or email address
+      const user = await userModel.findUserInput(userLogin);
+      if (!user) {
+        return res.status(404).json({
+          status: 'fail',
+          data: { message: 'Invalid Login Details' }
+        });
+      }
+      const passwordValid = await decryptPassword(password, user.password);
+      if (!passwordValid) {
+        return res.status(401).json({
+          status: 'fail',
+          data: { message: 'Invalid Login Details' }
+        });
+      }
+      const token = await generateToken(user);
+      return res.status(200).json({
+        status: 'success',
+        data: { token }
+      });
+    } catch (err) {
+      return next(err);
+    }
+  }
+
+  /**
+   *
+   * Forgot Password Middleware - Enables users request new password
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @returns {object} Object containing token to the user
+   * @memberof UsersController
+   */
+  static async forgotPassword(req, res, next) {
+    try {
+      const { userLogin } = req.body;
+      const user = await userModel.findUserInput(userLogin);
+      if (!user) {
+        return res.status(404).json({
+          status: 'fail',
+          data: { message: 'Invalid Login Details' }
+        });
+      }
+      const token = await crypto.randomBytes(20).toString('hex');
+      const mailSubject = 'Reset your password';
+      const mailContent = `You are receiving this mail because you requested
+       for a password change. Please Click on the link below or paste it
+       in a browser to complete the process <br />
+        http://${req.headers.host}/reset/${token} <br/>
+        Link expires in 1 hour time. <br />
+       If you did not request this, please Ignore this email and your
+       password will remain unchanged.
+      `;
+      await emailModule.sendEmailToUser(userLogin, mailSubject, mailContent);
```

Comment:
why are you awaiting the mail here on `line 110`?  this means the user will not get a response the mail is sent, what if there is a network error?

Reply:
I think the error has been captured by @abdulfatai360 when he created the Email module. I just imported and used it directly.

Reply:
okay